packages/plugins/experimental/plugin-wnfs/src/WnfsPlugin.tsx

```diff
@@ -58,3 +59,86 @@ export const WnfsPlugin = () =>
       activate: ReactSurface,
     }),
   ]);
+=======
+export const WnfsPlugin = (): PluginDefinition<WnfsPluginProvides> => {
+  let blockstore: Blockstore.MixedBlockstore | undefined;
+
+  return {
+    meta,
+    ready: async (plugins) => {
+      const client = resolvePlugin(plugins, parseClientPlugin)?.provides.client;
+      invariant(client);
+      const apiHost = client?.config?.values?.runtime?.services?.edge?.url || 'http://localhost:8787';
+      blockstore = Blockstore.create(apiHost);
+      await blockstore.open();
+    },
+    provides: {
+      translations,
+      metadata: {
+        records: {
+          [FileType.typename]: {
+            placeholder: ['file title placeholder', { ns: WNFS_PLUGIN }],
+            icon: (props: IconProps) => <FileCloud {...props} />,
+          },
+        },
+      },
+      echo: {
+        schema: [FileType],
+      },
+      // TODO(burdon): Add intent to upload file.
+      file: {
+        upload: async (file: File, space: Space) => {
+          if (!blockstore) {
+            throw new Error('Blockstore is not ready yet');
+          }
+
+          return await upload({ blockstore, file, space });
+        },
+      },
+      markdown: {
+        extensions: ({ document }: { document?: DocumentType }) => {
+          if (!blockstore) {
+            throw new Error('Blockstore is not ready yet');
+          }
+
+          if (document) {
+            const space = getSpace(document);
+            return space ? [imageExtension({ blockstore, space })] : [];
+          }
+
+          return [];
+        },
+      },
+      surface: {
+        component: ({ data, role, ...props }, forwardedRef) => {
+          // NOT USED IN COMPOSER VIEW
+
+          switch (role) {
+            case 'main':
+              return data.active instanceof FileType ? <FileMain file={data.active} /> : null;
+            case 'slide':
+              return data.slide instanceof FileType ? <FileSlide file={data.slide} cover={false} /> : null;
+            case 'section':
+              return data.object instanceof FileType ? <FileSection file={data.object} /> : null;
+            case 'card': {
+              if (
+                isObject(data.content) &&
+                typeof data.content.id === 'string' &&
+                data.content.object instanceof FileType
+              ) {
+                const cardProps = { ...props, item: { id: data.content.id, object: data.content.object } };
+                return isTileComponentProps(cardProps) ? (
+                  <FileCard {...cardProps} ref={forwardedRef as Ref<HTMLDivElement>} />
+                ) : null;
+              }
+              break;
+            }
+          }
+
+          return null;
+        },
+      },
+    },
+  };
+};
+>>>>>>> a8315dbb39 (feat: Track blocks that haven't been stored remotely yet)
```

Comment:
I think these can just be dropped. This was refactored so all the bits and pieces of this are in other files now.

Reply:
So fast! Just noticed I forgot this merge conflict sorry. Should've tried to build after I did the rebase on main before I pinged you 😅 In any case, the above should be resolved now, thanks!

Still doesn't build though, I'm getting a storybook error for some reason:
```
packages/sdk/app-framework/src/playground/playground.stories.tsx:9:39 - error TS2307: Cannot find module '@dxos/storybook-utils' or its corresponding type declarations.

9 import { withLayout, withTheme } from '@dxos/storybook-utils';
```

Reply:
that's weird... those are installed at the repo root I think so they should be available for all packages, but could try adding them to app-framework to see if it resolves the issue?

Reply:
Adding `@dxos/storybook-utils` to `app-framework` seems to work 👍 Everything should work as it's supposed to now.